server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/InternalHistogram.java

```diff
@@ -347,56 +348,75 @@ private double round(double key) {
     }
 
     private void addEmptyBuckets(List<Bucket> list, ReduceContext reduceContext) {
-        ListIterator<Bucket> iter = list.listIterator();
+        class Counter implements DoubleConsumer {
+            private int size = list.size();
+
+            @Override
+            public void accept(double key) {
+                size++;
+                if (size >= 10000) {
```

Comment:
I think this constant really needs an origin story, otherwise it makes the reader to wonder where it came from and if it is really meaningful. I wonder if it would make more sense to do something like DEFAULT_MAX_BUCKETS/10 with a brief explanation why this should work?

Reply:
That should work, yeah. I'll leave a nice comment. We could just keep adding and never call the method but that'd make failures slow. An origin story it is!